marketplacetests/tests/test_marketplace_in-app_not_you_login_feature.py

```diff
@@ -0,0 +1,71 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette import Wait
+
+from marketplacetests.firefox_accounts.app import FirefoxAccounts
+from marketplacetests.payment.app import Payment
+from marketplacetests.payment.regions.in_app import InAppPayment
+from marketplacetests.marketplace_gaia_test import MarketplaceGaiaTestCase
+from gaiatest.apps.homescreen.app import Homescreen
+from marketplacetests.marketplace.app import Marketplace
+from gaiatest.apps.homescreen.regions.confirm_install import ConfirmInstall
+
+
+class TestMakeInAppPayment(MarketplaceGaiaTestCase):
+
+    APP_NAME = 'Testing In-App-Payments'
+    APP_TITLE = 'In-App-Payments'
+
+    def setUp(self):
+        MarketplaceGaiaTestCase.setUp(self)
+        self.connect_to_local_area_network()
+
+        self.homescreen = Homescreen(self.marionette)
+
+        if not self.apps.is_app_installed(self.APP_NAME):
+
+            marketplace = Marketplace(self.marionette, self.MARKETPLACE_DEV_NAME)
+            marketplace.launch()
+
+            results = marketplace.search(self.APP_NAME)
+            results.install_searched_app(self.APP_NAME)
+
+            # Confirm the installation and wait for the app icon to be present
+            confirm_install = ConfirmInstall(self.marionette)
+            confirm_install.tap_confirm()
+
+            self.device.touch_home_button()
+
+        self.apps.switch_to_displayed_app()
+        self.homescreen.wait_for_app_icon_present(self.APP_NAME)
+
+    def test_make_an_in_app_payment(self):
+
+        username = self.testvars['marketplace']['username']
+        password = self.testvars['marketplace']['password']
+
+        # Verify that the app icon is visible on one of the homescreen pages
+        self.assertTrue(
+            self.homescreen.is_app_installed(self.APP_NAME),
+            'App %s not found on homescreen' % self.APP_NAME)
+
+        # Click icon and wait for h1 element displayed
+        self.homescreen.installed_app(self.APP_NAME).tap_icon()
+        Wait(self.marionette).until(lambda m: m.title == self.APP_TITLE)
+
+        tester_app = InAppPayment(self.marionette)
+        fxa = tester_app.tap_buy_product(is_logged_in=False)
+        fxa.login(username, password)
+
+        payment = Payment(self.marionette)
+        payment.tap_cancel_pin()
+
+        tester_app.tap_buy_product()
```

Comment:
This is confusing. In `tap_buy_product`, there is an argument called `is_logged_in` which defaults to `True`, and if so then returns `Payment`, otherwise it returns `FirefoxAccounts`. This is the only instance I can see where `tap_buy_product` is called and `is_logged_in` would be `True`, yet the code does not seem to expect `Payment` to be returned. In fact it does not interact with `Payment` at all from this point on, it only interacts with `fxa`.

This suggests there is something invalid/unnecessary with the `tap_buy_product` method.